gapis/replay/executor/executor.go

```diff
@@ -65,207 +60,91 @@ func Execute(
 	}.execute(ctx, connection)
 }
 
-func (e executor) execute(ctx context.Context, connection io.ReadWriteCloser) error {
-	// Encode the payload
-	// TODO: Make this a proto.
-	buf := &bytes.Buffer{}
-	w := endian.Writer(buf, e.memoryLayout.GetEndian())
-	w.Uint32(e.payload.StackSize)
-	w.Uint32(e.payload.VolatileMemorySize)
-	w.Uint32(uint32(len(e.payload.Constants)))
-	w.Data(e.payload.Constants)
-	w.Uint32(uint32(len(e.payload.Resources)))
-	for _, r := range e.payload.Resources {
-		w.String(r.ID)
-		w.Uint32(r.Size)
-	}
-	w.Uint32(uint32(len(e.payload.Opcodes)))
-	w.Data(e.payload.Opcodes)
-
-	data := buf.Bytes()
-
-	// Store the payload to the database
-	id, err := database.Store(ctx, data)
+func (e executor) execute(ctx context.Context, connection *gapir.Connection) error {
+	pp := e.payload.ToProto()
```

Comment:
Suggestion needed, this is making an extra copy?

Reply:
@ben-clayton  Please correct me if I am wrong, but I think there will be no extra copies. When you take the address of a go object, the compiler does its magic to make sure it will be on the heap, and no copy will be made.

Reply:
I don't know what the compiler will do, but I don't think you have to care.
Why are you worried? If it's regarding the payload data - they're in slices that will not be deep copied.